### ISSUE TITLE
Important performance drop when using medium texture quality

### ISSUE THREAD

**Issue by ad-daniel** [cyberbotics/webots#4343]
**Describe the Bug**

Set your OpenGL settings to:
- occlusion: disabled
- texture quality: **high**
- filtering: 4
- shadows: unchecked
- AA: unchecked

run:
`./webots projects/robots/dji/mavic/worlds/mavic_2_pro.wbt`
it takes about 10 seconds to load.

Now change the texture quality from  **high** to **medium**, close webots and run the same command again.
Now it takes ~44 seconds to load.

The effect varies from world to world, but the mavic is one particularly noticeable. What about it is the issue I'm not sure. either way, it doesn't seem a normal behavior when lowering the settings.

**Comment by omichel** [comment]
It could be that the down-scaling of the textures is taking time...

**Comment by ad-daniel** [comment]
Yes, it stays a long while stuck at 0% in the "finalizing nodes" state, which I assume is where that takes place.
Either way it seems excessive, and considering this world is also in the guided tour, having to wait for 45s for an overall very simple world seems too much. For reference, it takes 34 seconds to load CH_Vens. with the same settings.

**Comment by ShuffleWire** [comment]
I've been trying as well, and I could not experience the slowdown, even if I could says that visually the texture is well down-sampled (looking at the characters on the tape of the box, for instance) The world load within a consistent 10sec.
I have a not supported setup, however...
I'm still giving you feedback, hope it could help !

Webots : develop @ 0a240a599de5c8973fbbdf2214d803578ba5d2bd
OS : Debian 11
CPU : Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
GPU : None installed (driver not installed)
OpenGL informations :    
System: Linux 5.10.0-12-amd64 x86_64
OpenGL vendor: Intel
OpenGL renderer: Mesa Intel(R) UHD Graphics 630 (CFL GT2)
OpenGL version: 4.6 (Core Profile) Mesa 20.3.5
Available GPU memory: N/A

**Comment by ad-daniel** [comment]
Thanks! Did you compile in release mode or debug mode?

**Comment by ShuffleWire** [comment]
> Thanks! Did you compile in release mode or debug mode? (I did the test in debug)

Release

**Comment by ShuffleWire** [comment]
I've rebuilt in debug, and I could reproduce your issue ! got about 34sec on medium and 12sec on high (counting time in my head, not very accurate xD) But still, that seems reproducible

**Comment by ad-daniel** [comment]
Sorry, what I meant to say is that it stays stuck at 0% in the progress bar "finalizing nodes". not that the CPU is at 0% (quite the contrary), but you're right it's quite interesting that the low performs so well if indeed the issue is the down sampling

**Comment by BenjaminDeleze** [comment]
It seems that it is indeed the downscale that takes a lot of additional time:
For the mavic.wbt:
on low: 78 images, max 0.07 sec per images, total: between 3-5 seconds of overhead in this function
on medium: 69 images, max 0.3 sec per images, total between 12-15 seconds of overhead.
I will investigate further
